src/query/service/src/pipelines/processors/transforms/transform_sort_spill.rs

```diff
@@ -182,6 +182,7 @@ where
                         self.state = State::Merging;
                     }
                     self.input_data = Some(block);
+                    self.input.set_need_data();
```

Comment:
Maybe `self.input.finish()` is better if `meta.is_none()`

Reply:
Here we need to mark the last block to implement some optimisations